code/modules/food_and_drinks/kitchen_machinery/cooker.dm

```diff
@@ -111,6 +113,36 @@
 		to_chat(user, "<span class='warning'>Close the panel first!</span>")
 		return
 	if(!checkValid(I, user))
+		if(istype(I, /obj/item/weapon/grab))
+			var/obj/item/weapon/grab/G = I
+			if(istype(G.affecting, /mob/living/carbon/human) && (G.get_species != "Machine"))
+//				if(!iscarbon(G.affecting))
+//					to_chat(user, "<span class='warning'>You can only harm carbon-based creatures this way!</span>")
+//					return
+//				if(G.affecting.stat == DEAD)
+//					to_chat(user, "<span class='warning'>[G.affecting] is dead!</span>")
+//					return
+				if(G.state < GRAB_AGGRESSIVE)
+					to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+					return
+				var/mob/living/carbon/human/C = G.affecting
+				var/obj/item/organ/external/head/head = C.get_organ("head")
+				if(!head)
+					to_chat(user, "<span class='warning'>This person doesn't have a head!</span>")
+					return
+				if(istype(src, /obj/machinery/cooker/deepfryer))
```

Comment:
because deepfryer doesn't have it's own attackby proc call and in turn would give the annoying 'you can't insert this' message while you dunk their head in the deep fryer.